chef/data_bags/crowbar/template-network.json

```diff
@@ -184,6 +184,7 @@
           "add_bridge": false,
           "add_ovs_bridge": false,
           "bridge_name": "br-fixed",
+	  "ovs_forward_bpdu": false,
```

Comment:
There seem to be an indentation issue here.

Also I am wondering, since we marked the attribute optional. Should we really actually set a default here? Wouldn't it be better to rely on what ever ovs defaults to? Which means you also wouldn't need to do a schema migration. But in that case `Nic::OvsBridge.ovs_forward_bpdu(forward)` would need to learn about a third possible value `nil` which would mean, do whatever OVS does by default. @djoreilly  Any opinion on this?

Reply:
argh, will fix the indentation! (that's what happens when you paste from the file on your crowbar box at the end of the day...)  I think the field is just blank by default (at least, if you do `ovs-vsctl list Bridge` you won't see anything at all in the other_config section until you specify keys and values)-- but I don't know ovs really, just reporting what I see.

Reply:
> There seem to be an indentation issue here.
> 
> Also I am wondering, since we marked the attribute optional. Should we really actually set a default here? Wouldn't it be better to rely on what ever ovs defaults to? Which means you also wouldn't need to do a schema migration. But in that case `Nic::OvsBridge.ovs_forward_bpdu(forward)` would need to learn about a third possible value `nil` which would mean, do whatever OVS does by default. @djoreilly Any opinion on this?

yes, that seems cleaner. When OVS does not find `other_config:forward-bpdu` defined it uses false.

Reply:
I'm not sure why this wouldn't need a schema migration? Since the value hasn't been set from the barclamp so far, and aiui the user would be setting the value in the raw view of the barclamp. I may be misunderstanding something.

Reply:
You defined the attribute to be optional in the schema `{ "type": "bool", "required": false },` , which IMO is correct as we don't what every network to carry that attribute. But it means the validator doesn't require the attribute to be present. So you need to prepare the rest of your code to cope with proposals where the attribute is not set at all (even for the networks where you set a default, as the user could just delete it and the validator would not complain).

My proposal would be to not set a default anywhere and just document the new attribute in the deployment Guide `http://docserv.nue.suse.com/documents/SUSE_OpenStack_Cloud_9/suse-openstack-cloud-crowbar-deployment/single-html/#sec-deploy-network-json-networks` as I don't expect many setups to fiddle around with this setting.

When there is a network that doesn't have the `ovs_forward_bpdu` set to anything `BarclampLibrary::Barclamp::Inventory::Network.ovs_forward_bpdu` would return `nil` (I think). In that case the network cookbook just IMO just clear any existing value from the OVS database (and use what ever default OVS is using)

Only if the attribute is explicitly set in the proposal the cookbook should set it to either true of false.

Reply:
Ah, okay, gotcha. :) Thanks.